### PR TITLE
[WIP] Fix unassigned tuple that crashes pypy (and potentially CPython)

### PR DESCRIPTION
**Pull request by hmaarrfk**

### CODE REVIEW

File: cytoolz/itertoolz.pyx [hunk 1]
```diff
@@ -963,6 +963,7 @@ cdef class sliding_window:
         cdef Py_ssize_t i
         self.iterseq = iter(seq)
         self.prev = PyTuple_New(n)
+        PyTuple_SET_ITEM(self.prev, i, None)
```
Comment (mattip) [inline cytoolz/itertoolz.pyx:966]: Is `i` guaranteed to be `0` here? 
```suggestion
        PyTuple_SET_ITEM(self.prev, 0, None)
```

This is fine, cython will convert the `None` into `Py_None`, which is the C representation of `None`.

Comment (hmaarrfk) [reply cytoolz/itertoolz.pyx]: Got it, i didn't know if it would do the reference counting or not.

**Comment by eriknw** [comment]
LGTM!  Thanks so much @hmaarrfk and @mattip.  We should probably add `pypy` to the test matrix, but I can take care of that.  Merging.

**Comment by mattip** [comment]
It seems CI is running on travis-ci.org, which has transitioned to travis-ci.com and will be shutting down soonish. Some projects have moved to github actions/cibuildwheel to test and deploy, but they use githubs runners which are only x86 based (no arm64 yet).

**Pull request merged**